commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/BoundedEquivalentConcurrentHashMapV8.java

```diff
@@ -2737,27 +2725,33 @@ else if ((fh = f.hash) == MOVED)
                               oldVal = ev;
                               if (value != null) {
                                  e.val = value;
-                                 if (oldVal == null) {
-                                    evictionPolicy.onEntryMiss(e, value);
-                                 } else {
-                                    evictionPolicy.onEntryHitWrite(e, value);
+                                 if (!skipListener) {
+                                    if (oldVal == null) {
+                                       evictionPolicy.onEntryMiss(e, value);
+                                    } else {
+                                       evictionPolicy.onEntryHitWrite(e, value);
+                                    }
```

Comment:
These are getting big enough now that IntelliJ is showing me a duplicate code warning :)